intro_calculating_cred_grain.md

```diff
@@ -0,0 +1,162 @@
+
+#### Summary up to this point:
+
+In this way, PageRank takes weights in and outputs scores that value contributions based on their relationship to each other. The eigenvector <img src="https://render.githubusercontent.com/render/math?math=\vec v"> then consists of the Cred scores for the various nodes, including the contributors, so we’ve accomplished our goal!
+
+## Cred over time
+
+#### Time-based Cred constraints
+
+SourceCred has additionally innovated on PageRank to accommodate properties that Cred should demonstrate over time. The most important properties in this respect are that:
+
+-   a contributor should never lose cred. They ought to become more diluted as other contributors continue to earn new cred, but they’ll still continue to earn Grain on that Cred.
+-   a past action that was recently found to be valuable in the present should cause new Cred to flow to that action.
+
+#### Time epochs and fibration
+
+To simplify the computational overhead and storage required to compute Cred over time, SourceCred is moving toward a fibration model where a node is added per contributor, per unit time (currently weekly for SourceCred). In this model, epoch nodes intermediate flows to a user from any nodes during a period of time, and have a fixed transition probability to the user’s node, rather than a weight from which transition probabilities are calculated. The remaining transition probability is then shared between the seed node (for the teleportation vector) and the other node weights. This will be implemented by distinguishing graph edges, which have fixed weights, from markov edges, which have fixed transition probabilities from epoch nodes. This is then input into the mechanism that calculates the transition probabilities.
+
+## Grain dynamics
+
+#### How Grain is generated
+
+Once SourceCred calculates the Cred for individual contributors, that Cred is linked to an identity that is an amalgamation of their identities across the platforms (e.g. GitHub handle or Discord username). A constant amount of SourceCred Grain (¤) is produced per unit time, currently 15000¤ per week. This amount is distributed according to two different distribution mechanisms.
+
+-   20% of the grain is “fast,” and is disbursed to nodes in proportion to the cred earned by each node in the past week. This is intended to reward new participants.
+-   80% of the grain is “slow,” and is disbursed in proportion to the difference between total grain you've been distributed, and what that node’s proportional fair Cred would be over all history
+
+As a result, if you didn't do anything this week, you have only slow Grain. Alternatively, if you joined this week, you’ll only have fast Grain. The fast component can potentially over-distribute grain, but it will get diluted by other contributions.
+
+#### How Grain is traded:
```

Comment:
I think it should be added here that (the minted / actual / capitalized?) Grain is opt in - if you don't opt in, you don't need to worry about taxes, etc.

Reply:
I think this makes slightly more sense in the next section, and I'm changing the heading to "How much Grain is generated" so that it's a little more intuitive that the mechanics of earning grain belongs in the next section.

Reply:
I ended up softening the wording, since I don't want to make a claim as strong as "you don't have to worry about tax" in a doc that may not be updated before/when SourceCred moves Grain to an on-chain token.  Wouldn't want this to read like financial advice...

Reply:
That makes sense - I didn't mean to use those words verbatim, but for someone reading (who is a contributor) the message shouldn't be "Surprise! You get this grain stuff that you didn't ask for."

Reply:
awesome!  yeah, it's a great distinction to make.